### PR TITLE
feat: t6 - chapter2-l2-t6-get-file-unique-dirs.sh

### PR DESCRIPTION
**Pull request by darliiin**
**Summary**
**Test task**
t6 - get only dirs of the files
Script should get a folder as an argument and return a list of unique directories of all files (including in the nested directories). only -f option is allowed for the use of find command, i.e.: find /tmp -type f. Which should find all files in all nested dirs and then process the output by using expansion.

name the file: `chapter2-l2-t6-get-file-unique-dirs.sh`

**example of the call:**

`./chapter2-l2-t6-get-file-unique-dirs.sh ~/tmp`
```

/tmp/setguid
/tmp/bazel/tools/objc
/tmp/bazel/third_party/zlib
/tmp/bazel/third_party/zlib/test
/tmp/bazel/third_party/zlib
/tmp/bazel/third_party/zlib/examples
/tmp/bazel/third_party/zlib
/tmp/bazel/third_party/zlib/contrib/untgz
/tmp/bazel/third_party/zlib/contrib/testzlib
/tmp/bazel/third_party/zlib/contrib/puff

```
**my result**
`./chapter2-l2-t6-get-file-unique-dirs.sh ~/Загрузки/a
`
**output**
`/home/darlin/Загрузки/a/foto/photo_2023-03-20_16-00-58.jpg
/home/darlin/Загрузки/a/Maven_spring_demo.zip
/home/darlin/Загрузки/a/ideaIU-2022.3.3.tar.gz
/home/darlin/Загрузки/a/google-chrome-stable_current_amd64.deb
/home/darlin/Загрузки/a/2
/home/darlin/Загрузки/a/1
/home/darlin/Загрузки/a/3
/home/darlin/Загрузки/a/dariyanalimova.p12
`

### CODE REVIEW

File: chapter2/tests/t6/chapter2-l2-t6-get-file-unique-dirs.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+#Checking the number of arguments passed
```
[the remark this answers is not in the document]
Comment (darliiin) [reply chapter2/tests/t6/chapter2-l2-t6-get-file-unique-dirs.sh:2]: done

File: chapter2/tests/t6/chapter2-l2-t6-get-file-unique-dirs.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (darliiin) [reply chapter2/tests/t6/chapter2-l2-t6-get-file-unique-dirs.sh:1]: done

File: chapter2/tests/t6/chapter2-l2-t6-get-file-unique-dirs.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (kseniyashaydurova) [reply chapter2/tests/t6/chapter2-l2-t6-get-file-unique-dirs.sh]: Output is still incorrect
image

**Review by kseniyashaydurova — CHANGES_REQUESTED** [review]
Output is still incorrect 
image

**Review by roman-mightydevops — APPROVED** [review]

**Review by kseniyashaydurova — APPROVED** [review]

**Pull request merged**